rmgpy/data/solvation.py

```diff
@@ -1878,71 +2011,70 @@ def get_Kfactor_parameters(self, solute_data, solvent_data, T_trans_factor=0.75)
-            rho_l_T2 = PropsSI('Dmolar', 'T', T2, 'Q', 0, solvent_name)
-            rho_g_T2 = PropsSI('Dmolar', 'T', T2, 'Q', 1, solvent_name)
-            K_T2 = math.exp(delG_T2 / (T2 * constants.R)) / rho_g_T2 * rho_l_T2
-            x_T2 = T2 / Tc * math.log(K_T2)  # Tln(K-factor) at 299 K, in K
-            slope298 = (x_T2 - x298) / (T2 - 298)
-            Amatrix[1][0] = 0
-            Amatrix[1][1] = -0.355 / Tc * ((1 - 298 / Tc) ** (-0.645))
-            Amatrix[1][2] = 1 / Tc * math.exp(1 - 298 / Tc) * (0.59 * (298 / Tc) ** (-0.41) - (298 / Tc) ** 0.59)
-            Amatrix[1][3] = 0
-            bvec[1] = slope298
-            # 3. Tln(K-factor) continuity at T = T_transition
-            rho_l_Ttran = PropsSI('Dmolar', 'T', T_transition, 'Q', 0, solvent_name)
-            Amatrix[2][0] = 1
-            Amatrix[2][1] = (1 - T_transition / Tc) ** 0.355
-            Amatrix[2][2] = math.exp(1 - T_transition / Tc) * (T_transition / Tc) ** 0.59
-            Amatrix[2][3] = -(rho_l_Ttran - rho_c) / rho_c
-            bvec[2] = 0
-            # 4. d(Tln(K-factor)) / dT smooth transition at T = T_transition
-            T3 = T_transition + 1
-            rho_l_T3 = PropsSI('Dmolar', 'T', T3, 'Q', 0, solvent_name)
-            Amatrix[3][0] = 0
-            Amatrix[3][1] = -0.355 / Tc * ((1 - T_transition / Tc) ** (-0.645))
-            Amatrix[3][2] = 1 / Tc * math.exp(1 - T_transition / Tc) * (
-                        0.59 * (T_transition / Tc) ** (-0.41) - (T_transition / Tc) ** 0.59)
-            Amatrix[3][3] = - ((rho_l_T3 - rho_l_Ttran) / rho_c / (T3 - T_transition))
-            bvec[3] = 0
-            # solve for the parameters
-            param, residues, ranks, s = np.linalg.lstsq(Amatrix, bvec, rcond=None)
-            # store the results in kfactor_parameters class
-            kfactor_parameters = KfactorParameters()
-            kfactor_parameters.lower_T = [float(param[0]), float(param[1]), float(param[2])]
-            kfactor_parameters.higher_T = float(param[3])
-            kfactor_parameters.T_transition = T_transition
-        else:
-            raise DatabaseError("K-factor parameter calculation is not available for the solvent "
-                                "whose `name_in_coolprop` is None")
+
+        Args:
+            delG298 (float): solvation free energy at 298 K in J/mol.
+            delH298 (float): solvation enthalpy at 298 K in J/mol.
+            delS298 (float): solvation entropy at 298 K in J/mol/K.
+            solvent_name (str): name of the solvent that is used in CoolProp.
+            T_trans_factor (float): a temperature
+
+        """
+
+        Tc = get_critical_temperature(solvent_name)
+        T_transition = Tc * T_trans_factor  # T_trans_factor is empirically set to 0.75 by default
+        rho_c = PropsSI('rhomolar_critical', solvent_name)  # the critical density of the solvent, in mol/m^3
+
+        # Generate Amatrix and bvector for Ax = b
```

Comment:
The previous code for this section had error catching for when the solvent wasn't in the database. Is this needed here as well?

Reply:
I removed that section because that error will be caught in the first line when it runs `Tc = get_critical_temperature(solvent_name)`. So I think it's redundant.